### ISSUE TITLE
uninitialized constant Doorkeeper::JWT::JWK in 1.8.4

### ISSUE THREAD

**Issue by brent-cybrid** [doorkeeper-gem/doorkeeper-openid_connect#187]
I just upgraded to `doorkeeper-openid_connect` version 1.8.4 and my application is no longer running with the following error and stack trace:

```
JWT::JWK.new(key)
       ^^^^^> with backtrace:

# /Users/user/.gem/ruby/3.1.3/gems/doorkeeper-openid_connect-1.8.4/lib/doorkeeper/openid_connect.rb:51:in `signing_key'
# /Users/user/.gem/ruby/3.1.3/gems/doorkeeper-openid_connect-1.8.4/lib/doorkeeper/openid_connect/id_token.rb:35:in `as_jws_token'
```

**Comment by nbulaj** [comment]
And with 1.8.3 it works? Could be related to https://github.com/doorkeeper-gem/doorkeeper-openid_connect/pull/177 ?

**Comment by nbulaj** [comment]
Yeah so to fix it there are two ways:
* Migrate to the new gem
* Fix doorkeeper-openid_connect to support both gems (maybe @kristof-mattei will want to check)

**Comment by zavan** [comment]
I have the exact same issue but my project does not rely on `json-jwt` at all, it's not even in my Gemfile.lock.

In my Gemfile I have:

```ruby
gem 'doorkeeper'
gem 'doorkeeper-jwt'
gem 'doorkeeper-openid_connect'
```

All in the latest version.

Gemfile.lock:

```
    doorkeeper (5.6.4)
      railties (>= 5)
    doorkeeper-jwt (0.4.1)
      jwt (>= 2.1)
    doorkeeper-openid_connect (1.8.4)
      doorkeeper (>= 5.5, < 5.7)
      jwt (>= 2.5)
    oauth2 (2.0.9)
      faraday (>= 0.17.3, < 3.0)
      jwt (>= 1.0, < 3.0)
    ...
    jwt (2.6.0)
```

When I try to access the `JWT::JWK` class in the Rails console it works:

```
irb(main):004:0> JWT::JWK
=> JWT::JWK
irb(main):005:0> JWT::JWK.method(:new)
=> #<Method: JWT::JWK.new(create_from)(key, params=..., options=...) /usr/local/bundle/gems/jwt-2.6.0/lib/jwt/jwk.rb:9>
```

**Comment by zavan** [comment]
Found the problem, PR to follow.

**Comment by kristof-mattei** [comment]
@zavan your error is different. When you have the error it is searching for `Doorkeeper::JWT::JWK`, which is in correct. I think we can fix your error by prefixing https://github.com/doorkeeper-gem/doorkeeper-openid_connect/blob/b268d28c61f48fdacd5675fefb863b823d91d515/lib/doorkeeper/openid_connect.rb#L51 with `::` to ensure it doesn't search for `JWT` in the same namespace (i.e. `Doorkeeper`). This also explains why it works in the console. 

Can you try changing that file and confirm if that does the trick? 

@brent-cybrid can you post the complete stacktrace? Want to make sure that it is the gem's absence and not something I missed. 

@nbulaj I think supporting both is another can of worms, which I find quite risky when it comes to security-related code like this. It's not even like we can specify the model the consumer would like to use, like https://doorkeeper.gitbook.io/guides/configuration/models, as the APIs are actually different. Thoughts?

**Comment by zavan** [comment]
@kristof-mattei Yeah, I tested #188 and it worked.

**Comment by nbulaj** [comment]
Released with `1.8.5`
